src/App.tsx

```diff
@@ -1,12 +1,66 @@
 import React from 'react';
 import './App.scss';
+import GoodsList from './GoodsList/GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Product } from './types/Product';
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+type State = {
+  products: Product[] | null
+};
 
-export default App;
+export class App extends React.Component<{}, State> {
+  state: State = {
+    products: null,
+  };
+
+  getData = async (func: () => Promise<Good[]>) => {
+    try {
+      const goods = await func();
+
+      this.setState({ products: goods });
+    } catch (error) {
+      throw new Error(`Error - ${error}`);
```

Comment:
And the execution stops cause you have unhandled error, you can show some kind of message to user if there was an error

Reply:
Just remove throwing this error
```suggestion
```